### PR TITLE
ROX-24948: simple script for getting external entities from Central

### PR DESCRIPTION
**Pull request by Stringy**
Script for interacting with external IPs, based on network graph API changes in stackrox/stackrox#12689

### CODE REVIEW

**Review by mfosterrox — APPROVED** [review]
lgtm

File: util-scripts/external-entities/external-entities.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+        response = requests.get(self._endpoint_path(path), params=params, headers=headers, verify=False).json()
+        self._handle_error_response(response)
+        return response
+
+    def _endpoint_path(self, path):
+        return f'https://{self._auth.endpoint}/{path}'
+
+    def _handle_error_response(self, response):
+        if 'error' not in response:
```
Comment (Molter73) [inline util-scripts/external-entities/external-entities.py:89]: Does this imply the API might return 200 with an error? Why are we not checking the status code?

File: util-scripts/external-entities/external-entities.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+
+class Client:
+    def __init__(self, cluster, auth):
+        self._auth = auth
+        self._cluster = cluster
+        self._cluster_id = None
+
+    def get_all_external_entities(self, learned=True, cidr=None):
+        query = f"Learned External Source:{str(learned).lower()}"
```
[the remark this answers is not in the document]
Comment (Stringy) [reply util-scripts/external-entities/external-entities.py:34]: Yeah this is the custom query language used by the search framework - it's translated into SQL WHERE clauses by central before querying the database. There's not much documentation on this but have a look around https://github.com/stackrox/stackrox/blob/master/pkg/search and https://github.com/stackrox/stackrox/tree/master/pkg/search/postgres/query if you're interested :)

File: util-scripts/external-entities/external-entities.py (hunk 1 above)
Comment (Stringy) [reply util-scripts/external-entities/external-entities.py:89]: Ya know what I didn't even think to look at the status code, I just saw it was returning json and left it at that. I'll check and fix.

File: util-scripts/external-entities/requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+tabulate
```
[the remark this answers is not in the document]
Comment (Stringy) [reply util-scripts/external-entities/requirements.txt:1]: I could have sworn requests was a standard library in python3. I'll update the requirements

[the remark this answers is not in the document]
Comment (Molter73) [reply util-scripts/external-entities/requirements.txt]: > I could have sworn requests was a standard library in python3. I'll update the requirements

It should, but nope, everyone installs it at some point though.

**Pull request merged**